### PR TITLE
e2e, network: Unify IP report in status tests

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Before this PR:

Currently there are two tests that verifies VMI with bridge binding report the pod IP in status at test/network/primary_pod_network.go.
Both tests use Alpine VM which doesnt have QEMU guest-agent installed.

After this PR:
The are unified into the "multiple VM with bridge binding" test at tests/network/vmi_networking.go.
This test use Cirros VM which doesnt have QEMU guest-agent installed as well.

This PR change tests that perform connectivity check cross nodes to not skip silently but fail loudly, with a informative message.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/network/vmi_networking.go [hunk 1]
```diff
@@ -176,6 +176,8 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 				}
 				Expect(libnet.CheckMacAddress(vmi, "eth0", vmi.Status.Interfaces[0].MAC)).To(Succeed())
 
+				By("check IP/s reported on the VMI matches the ones on the pod")
+				Expect(libnet.ValidateVMIandPodIPMatch(vmi, vmiPod)).To(Succeed(), "Should have matching IP/s between pod and vmi")
 			},
 				Entry("[test_id:1539]the Inbound VirtualMachineInstance", &inboundVMI),
```
Comment (orelmisan) [inline tests/network/vmi_networking.go:182]: Please consider stating that the VMI is expected to implicitly have the default binding -
Which is also implicitly assumed to be `bridge` since it is not explicitly set in the KubeVirt CR.
https://github.com/kubevirt/kubevirt/blob/f68fb96c0e70b818b2b0c606aab35f68c9eb462d/pkg/virt-config/virt-config.go#L58

File: examples/vmi-alpine-efi.yaml [hunk 2]
```diff
@@ -12,10 +12,6 @@ spec:
       - disk:
           bus: virtio
         name: containerdisk
-    firmware:
-      bootloader:
-        efi:
-          secureBoot: false
```
Comment (orelmisan) [inline examples/vmi-alpine-efi.yaml:18]: I don't follow why was this file changed, could you please elaborate?

File: tests/network/primary_pod_network.go [hunk 3]
```diff
@@ -105,21 +105,6 @@ var _ = SIGDescribe("Primary Pod Network", func() {
 					Eventually(vmiIPs, 2*time.Minute, 5*time.Second).Should(Equal(podIPs), "should contain VMI Status IP as Pod status IPs")
 				})
 			})
-			When("no Guest Agent exists", func() {
```
Comment (orelmisan) [inline tests/network/primary_pod_network.go:118]: The commit message states:
> The original test use Alpine VM that has no QEMU-GA installed, the target test
use Cirros VM which doesn't have QEMU-GA installed as well.

How about:
> Both the test removed by this commit and `[test_id:1540]` use OS images that do not contain QEMU-GA.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @ormergi.
Please see the comments inline.

File: tests/network/vmi_networking.go (hunk 1 above)
Comment (ormergi) [reply tests/network/vmi_networking.go:182]: It will not be correct for all cases, see bellow entries, there are three different VM configurations.
The first one is a VMI with no network/ifaces configured.

File: examples/vmi-alpine-efi.yaml (hunk 2 above)
Comment (ormergi) [reply examples/vmi-alpine-efi.yaml:18]: Its not related, sorry for the noise.
Fixed.

File: tests/network/primary_pod_network.go (hunk 3 above)
Comment (ormergi) [reply tests/network/primary_pod_network.go:118]: Done

File: tests/network/vmi_networking.go (hunk 1 above)
Comment (ormergi) [reply tests/network/vmi_networking.go:182]: Done

**Review by orelmisan — APPROVED** [review]
Thank you @ormergi

File: tests/network/primary_pod_network.go (hunk 3 above)
Comment (EdDev) [reply tests/network/primary_pod_network.go:118]: nice to have: It would have been easier to track the alternative test using a permanent link (from the last tagged release).

**Comment by ormergi** [comment]
> Thank you for removing two e2e tests.
> 
> Please adjust the commit messages to fit the recommended line length [1].
> 
> [1] https://cbea.ms/git-commit/#wrap-72

Done
PTAL

**Review by EdDev — COMMENTED** [review]
Thanks!

/approve

**Review by orelmisan — APPROVED** [review]
